### ISSUE TITLE
chilled_log_softmax

### ISSUE THREAD

**Issue by RaptorMai** [soochan-lee/CN-DPM#6]
Firstly thank you so much for sharing the codebase. In the function `nll(self, x, y, step=None)`, you calculate the chilled_log_softmax. I am wondering what's the meaning of it and why do you want to use it.  Thank you in advance. 

```python
        # Classifier chilling
        chilled_log_softmax = F.log_softmax(
            log_softmax / self.config['classifier_chill'], dim=1)
        chilled_loss_pred = self.ce_loss(chilled_log_softmax, y)

        # Value with chill & gradient without chill
        loss_pred = loss_pred - loss_pred.detach() \
            + chilled_loss_pred.detach()
```

**Comment by soochan-lee** [comment]
Thank you for your interest in our work!

Because the dimension of the input is much bigger than the output, the scale of the VAE loss is far greater than the classifier loss. Due to this scale difference, the classifier loss is almost completely ignored during expert selection if not properly handled. So we lower the temperature of the classifier output (hence the name `classifier_chill`) to match the loss scale of the classifier. However, if you lower the temperature during training, the chilled prediction still behaves like a normal prediction, while the gradients explode. That's why we compute the gradients with the normal prediction, while outputting the chilled prediction.
